services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -1593,4 +1606,36 @@ private MacAddress getPortHwAddress(IOFSwitch sw, int portNumber) throws SwitchO
     private boolean isDefaultRule(long cookie) {
         return (cookie & DEFAULT_RULES_MASK) != 0L;
     }
+
+    private boolean meterExists(IOFSwitch sw, long meter, long rate) {
+        try {
+            verifySwitchSupportsMeters(sw);
+            OFMeterConfig meterConfig = getMeter(sw.getId(), meter);
+            long expectedRate = isSwitchSupportsPktpsFlag(sw) ? rate : getRateForMeterInKbps(rate);
+
+            List<OFMeterBand> ofMeterBands = meterConfig.getEntries();
+            if (ofMeterBands == null || ofMeterBands.isEmpty()) {
+                return false;
+            }
+
+            OFMeterBand meterBand = ofMeterBands.get(0);
+            if (meterBand instanceof OFMeterBandDrop) {
+                OFMeterBandDrop meterBandDrop = (OFMeterBandDrop) meterBand;
+                return meterBandDrop.getRate() == expectedRate;
+            }
+        } catch (SwitchOperationException e) {
+            logger.warn(format("Failed to verify the existence of meter %s", meter), e);
+        }
+
+        return false;
+    }
+
+    private long getRateForMeterInKbps(long ratePkts) {
+        return (ratePkts * config.getDiscoPacketSize()) / 1024L;
```

Comment:
Why it is into config? The best way is to "ask" it from `PathVerificationModule`, but it will(probably) lead to circular modules dependency.